chain-abci/src/enclave_bridge/edp/mod.rs

```diff
@@ -22,36 +22,55 @@ use tokio::net::{TcpListener, TcpStream};
 
 use std::os::unix::net::UnixStream;
 
+const REMOTE_ATTESTATION_PROXY: &str = "ra-sp-server";
+
 /// pair of unix domain sockets
 /// enclave_stream is only needed / passed in `connect_stream`
 /// `runner_stream` is shared in chain-abci app
 #[derive(Debug)]
 pub struct TxValidationApp {
     enclave_stream: Option<UnixStream>,
     runner_stream: Arc<Mutex<UnixStream>>,
+    /// `ra-sp-server` address for remote attestation. E.g. `0.0.0.0:8989`
+    /// FIXME: enclave direct connection -- not via TCP proxy
+    sp_address: Option<String>,
+    tdbe_stream: Option<UnixStream>,
 }
 
 impl Clone for TxValidationApp {
+    /// clone is only used for `TxValidationServer`/`chain-abci` having access to "runner_stream",
+    /// not for launching
     fn clone(&self) -> Self {
         Self {
             enclave_stream: None,
             runner_stream: self.runner_stream.clone(),
+            sp_address: None,
+            tdbe_stream: None,
         }
     }
```

Comment:
`Clone` is not the right trait to implement for this purpose. Maybe have a function which returns `self.runner_stream.clone()`?

Reply:
yes, this is a relic of older code, so clone doesn't make sense much. I'll put it in a different function